oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/JdbcOAuth2AuthorizationService.java

```diff
@@ -551,6 +610,25 @@ private String writeMap(Map<String, Object> data) {
 
 	}
 
+	public static class DefaultColumnTypeExtractor implements Function<JdbcOperations, Integer> {
+
+		@Override
+		public Integer apply(JdbcOperations jdbcOperations) {
+			return jdbcOperations.execute((ConnectionCallback<Integer>) con -> {
+				DatabaseMetaData databaseMetaData = con.getMetaData();
+				ResultSet rs = databaseMetaData.getColumns(null, null, TABLE_NAME, "access_token_value");
+				if (rs.next()) {
+					return rs.getInt("DATA_TYPE");
+				}
+				rs = databaseMetaData.getColumns(null, null, TABLE_NAME.toUpperCase(), "access_token_value".toUpperCase());
```

Comment:
What's the reason for this 2nd call using `toUpperCase()`?

Reply:
hi @jgrandja . Based on [HSQLDBdocs ](http://hsqldb.org/doc/guide/databaseobjects-chapt.html#N10E90)
```
When a database object is created with one of the CREATE statements or renamed with the ALTER statement, if the name is enclosed in double quotes, the exact name is used as the case-normal form. But if it is not enclosed in double quotes, the name is converted to uppercase and this uppercase version is stored in the database as the case-normal form.
```

`oauth2-authorization-schema.sql` is defining the table name in lower case but when reading the metadata the table name is in upper case -> the getColumns is returning 0 results.

Reply:
Gotchya. Can you put a `NOTE:` in the code so we know the reason. Let's leave it for now to get this merged and maybe do some cleanup later.